### DISCUSSION TITLE
Automate the plugin content in Docusaurus.config.js after adding a new MD folder

### DISCUSSION THREAD

**Discussion by SudhaM-Hexaware** [facebook/docusaurus#7948] [v1-v2 migration]
I am trying to automate the process of adding the MD files in docusaurus. Thereby need to define values in **plugins** - **Docusaurus.config.js** each time.   Are there any ways to automate the plugin content after adding a new MD folder? 

`plugin=[[
      "@docusaurus/plugin-content-docs",
      {
        id: "reacthooks",
        path: "reacthooks",
        routeBasePath: "reacthooks",
        sidebarPath: require.resolve("./sidebars.js"),
      },
    ]`

I want the above lines to be automatically added to Docusaurus. config file whenever a new MD folder gets added.

**Comment by slorber** [comment]
The config file is Node.js so technically you can use whatever you want here to create your config dynamically based on your own file-system conventions. 

What prevents you from doing this now yourself: use node fs, read the filesystem and compute a dynamic array of docs plugins?

**Reply by SudhaM-Hexaware** [reply]
The function to fetch data from API has to run first before exporting config or plugins. And, where should I call the fetch function in the config file? I tried many alternative methods by importing from other files. Still, **unable to use FETCH operation inside Docusaurus.config.js**

**Reply by slorber** [reply]
See https://github.com/facebook/docusaurus/discussions/7847#discussioncomment-3355923

edit: oh ok you are the same user 😅 Then I don't know what else to say, I gave you an example of using fetch inside Docusaurus.config.js already 🤷‍♂️  Do you have enough knowledge of nodejs and how async code works? What doesn't work exactly, show me the code (not words), error messages etc...
